src/api/code-challenge/content-types/code-challenge/test-evaluator.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  evaluateWithContext,
+  getCode,
+  getEvaluationContext,
+  overrideConsoleLog,
+  restoreConsoleLog,
+} from "../code-challenge/utils";
+
+type runTestEvaluatorProps = {
+  metaCaseCode: string;
+  internalTestCode: string;
+  metaLabel: string;
+  challengeLabel: string;
+  metaTestId: number;
+  evalResultShouldBe: boolean;
+  internalTest: { id: number };
```

Comment:
I don't understand. Instead of doign this confusing thing where you rename `internalTest` to `internalTestCode` why not just do 
```
internalTest: {
  id: number;
  code: string;
}
```

Reply:
Because internalTest is originally defined in database, which strapi is searching. When strapi returns the internalTest, it is a string.

In strapi it is:
code-challenge-test = {
  internalTest : "some code here",
  id : "a number value",
  label : "title of the internalTest"
}

```async function getInternalTests(eventTests) {
  const internalTests = await strapi.db
    .query("challenge.code-challenge-test")
    .findMany({
      select: ["id", "internalTest", "label"],
      where: {
        id: eventTests.map(({ id: testId }) => testId),
      },
    });```